block/produce.go

```diff
@@ -66,12 +69,28 @@ func (m *Manager) ProduceBlockLoop(ctx context.Context) {
 				continue
 			}
 			resetEmptyBlocksTimer()
-		case shouldProduceBlocks := <-m.shouldProduceBlocksCh:
-			for !shouldProduceBlocks {
-				m.logger.Info("block production paused - awaiting positive continuation signal")
-				shouldProduceBlocks = <-m.shouldProduceBlocksCh
+
+			// Check if we should submit the accumulated data
+			if m.shouldSubmitBatch() {
+				select {
+				case m.ShouldSubmitBatchCh <- true:
+					m.logger.Info("new batch accumualted, signal sent to submit the batch")
+				default:
+					m.logger.Error("new batch accumualted, but channel is full, stopping block production until the signal is consumed")
+					// emit unhealthy event for the node
+					evt := &events.DataHealthStatus{Error: fmt.Errorf("submission channel is full")}
+					uevent.MustPublish(ctx, m.Pubsub, evt, events.HealthStatusList)
+					// wait for the signal to be consumed
+					m.ShouldSubmitBatchCh <- true
+					m.logger.Info("resumed block production")
+					// emit healthy event for the node
+					evt = &events.DataHealthStatus{Error: nil}
+					uevent.MustPublish(ctx, m.Pubsub, evt, events.HealthStatusList)
+				}
+				m.produceBlockMutex.Lock()
+				m.AccumulatedProducedSize.Store(0)
```

Comment:
IMO the AccumulatedProducedSize should be set to 0 where the batch is created, otherwise it is only set to 0 after submitting a batch when reaching the max size, but not after submitting a batch after reaching the max time.

Reply:
As I see it, once enough data produced, and a submit signal is sent, the produced should start accumulating from 0.
otherwise, the producer needs to know and handle the submission state, which I try to avoid.

I'll think of solution for this case

Reply:
This is a "low traffic" edge case.
In the worst case, a half-empty batch will be sent after previous half empty batch was sent.

I'll open separate issue for it

Reply:
im not sure if i get what you mean, but, unless im wrong regarding the code, I think the problem is not setting to 0 it here, or knowing the submission state. i think the problem is not setting it to 0 after a batch is produced because it reached the maximum time. i think it should start accumulating size since last batch was created, no matter if it was triggered by max size or max time. otherwise it can happen that a batch is created because the maximum time is reached, and just after a few blocks another batch is created because the maximum batch size is reached, but the batch size will be actually only those few blocks.